Build issue with clang8 on amd64

```
In file included from src/secp256k1.c:12:
In file included from ./src/scalar_impl.h:20:
./src/scalar_4x64_impl.h:379:19: error: value '4624529908474429119' out of
range for constraint 'n'
    : "S"(l), "n"(SECP256K1_N_C_0), "n"(SECP256K1_N_C_1)
		  ^~~~~~~~~~~~~~~
./src/scalar_4x64_impl.h:17:25: note: expanded from macro 'SECP256K1_N_C_0'
#define SECP256K1_N_C_0 (~SECP256K1_N_0 + 1)
			^~~~~~~~~~~~~~~~~~~~
./src/scalar_4x64_impl.h:458:74: error: value '4624529908474429119' out of
range for constraint 'n'
    : "g"(m0), "g"(m1), "g"(m2), "g"(m3), "g"(m4), "g"(m5), "g"(m6),
"n"(SECP256K1_N_C_0), "n"(SECP256K1_N_C_1)
									
^~~~~~~~~~~~~~~
./src/scalar_4x64_impl.h:17:25: note: expanded from macro 'SECP256K1_N_C_0'
#define SECP256K1_N_C_0 (~SECP256K1_N_0 + 1)
			^~~~~~~~~~~~~~~~~~~~
./src/scalar_4x64_impl.h:504:64: error: value '4624529908474429119' out of
range for constraint 'n'
    : "g"(p0), "g"(p1), "g"(p2), "g"(p3), "g"(p4), "D"(r),
"n"(SECP256K1_N_C_0), "n"(SECP256K1_N_C_1)
							       ^~~~~~~~~~~~~~~
./src/scalar_4x64_impl.h:17:25: note: expanded from macro 'SECP256K1_N_C_0'
#define SECP256K1_N_C_0 (~SECP256K1_N_0 + 1)
			^~~~~~~~~~~~~~~~~~~~
```

I can reproduce this on debian sid with

```sh
$ clang-8 --version
clang version 8.0.0-+rc3-1~exp1 (tags/RELEASE_800/rc3)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin

$ ./autogen.sh && ./configure CC=clang-8 && make
...
Build Options:
  with endomorphism   = no
  with ecmult precomp = yes
  with jni            = no
  with benchmarks     = yes
  with coverage       = no
  module ecdh         = no
  module recovery     = no

  asm                 = x86_64
  bignum              = no
  field               = 64bit
  scalar              = 64bit

  CC                  = clang-8
  CFLAGS              = -g -O2 -W -std=c89 -pedantic -Wall -Wextra -Wcast-align -Wnested-externs -Wshadow -Wstrict-prototypes -Wno-unused-function -Wno-long-long -Wno-overlength-strings -fvisibility=hidden -O3
  CPPFLAGS            = 
  LDFLAGS             = 

  CC       src/bench_verify.o
gcc -I. -g -O2 -Wall -Wextra -Wno-unused-function -c src/gen_context.c -o gen_context.o
gcc -g -O2 -Wall -Wextra -Wno-unused-function  gen_context.o -o gen_context
./gen_context
  CC       src/libsecp256k1_la-secp256k1.lo
In file included from src/secp256k1.c:12:
In file included from ./src/scalar_impl.h:20:
./src/scalar_4x64_impl.h:379:19: error: value '4624529908474429119' out of range for constraint 'n'
    : "S"(l), "n"(SECP256K1_N_C_0), "n"(SECP256K1_N_C_1)
                  ^~~~~~~~~~~~~~~
./src/scalar_4x64_impl.h:17:25: note: expanded from macro 'SECP256K1_N_C_0'
#define SECP256K1_N_C_0 (~SECP256K1_N_0 + 1)
                        ^~~~~~~~~~~~~~~~~~~~
./src/scalar_4x64_impl.h:458:74: error: value '4624529908474429119' out of range for constraint 'n'
    : "g"(m0), "g"(m1), "g"(m2), "g"(m3), "g"(m4), "g"(m5), "g"(m6), "n"(SECP256K1_N_C_0), "n"(SECP256K1_N_C_1)
                                                                         ^~~~~~~~~~~~~~~
./src/scalar_4x64_impl.h:17:25: note: expanded from macro 'SECP256K1_N_C_0'
#define SECP256K1_N_C_0 (~SECP256K1_N_0 + 1)
                        ^~~~~~~~~~~~~~~~~~~~
./src/scalar_4x64_impl.h:504:64: error: value '4624529908474429119' out of range for constraint 'n'
    : "g"(p0), "g"(p1), "g"(p2), "g"(p3), "g"(p4), "D"(r), "n"(SECP256K1_N_C_0), "n"(SECP256K1_N_C_1)
                                                               ^~~~~~~~~~~~~~~
./src/scalar_4x64_impl.h:17:25: note: expanded from macro 'SECP256K1_N_C_0'
#define SECP256K1_N_C_0 (~SECP256K1_N_0 + 1)
                        ^~~~~~~~~~~~~~~~~~~~
3 errors generated.
make: *** [Makefile:1047: src/libsecp256k1_la-secp256k1.lo] Error 1

--- a/src/scalar_4x64_impl.h
+++ b/src/scalar_4x64_impl.h
@@ -376,7 +376,7 @@ static void secp256k1_scalar_reduce_512(secp256k1_scalar *r, const uint64_t *l) 
     /* extract m6 */
     "movq %%r8, %q6\n"
     : "=g"(m0), "=g"(m1), "=g"(m2), "=g"(m3), "=g"(m4), "=g"(m5), "=g"(m6)
-    : "S"(l), "n"(SECP256K1_N_C_0), "n"(SECP256K1_N_C_1)
+    : "S"(l), "i"(SECP256K1_N_C_0), "i"(SECP256K1_N_C_1)
     : "rax", "rdx", "r8", "r9", "r10", "r11", "r12", "r13", "r14", "cc");
 
     /* Reduce 385 bits into 258. */
@@ -455,7 +455,7 @@ static void secp256k1_scalar_reduce_512(secp256k1_scalar *r, const uint64_t *l) 
     /* extract p4 */
     "movq %%r9, %q4\n"
     : "=&g"(p0), "=&g"(p1), "=&g"(p2), "=g"(p3), "=g"(p4)
-    : "g"(m0), "g"(m1), "g"(m2), "g"(m3), "g"(m4), "g"(m5), "g"(m6), "n"(SECP256K1_N_C_0), "n"(SECP256K1_N_C_1)
+    : "g"(m0), "g"(m1), "g"(m2), "g"(m3), "g"(m4), "g"(m5), "g"(m6), "i"(SECP256K1_N_C_0), "i"(SECP256K1_N_C_1)
     : "rax", "rdx", "r8", "r9", "r10", "r11", "r12", "r13", "cc");
 
     /* Reduce 258 bits into 256. */
@@ -501,7 +501,7 @@ static void secp256k1_scalar_reduce_512(secp256k1_scalar *r, const uint64_t *l) 
     /* Extract c */
     "movq %%r9, %q0\n"
     : "=g"(c)
-    : "g"(p0), "g"(p1), "g"(p2), "g"(p3), "g"(p4), "D"(r), "n"(SECP256K1_N_C_0), "n"(SECP256K1_N_C_1)
+    : "g"(p0), "g"(p1), "g"(p2), "g"(p3), "g"(p4), "D"(r), "i"(SECP256K1_N_C_0), "i"(SECP256K1_N_C_1)
     : "rax", "rdx", "r8", "r9", "r10", "cc", "memory");
 #else
     uint128_t c;

